utilities/checkpoint/checkpoint_impl.cc

```diff
@@ -46,28 +46,38 @@ Status Checkpoint::CreateCheckpoint(const std::string& /*checkpoint_dir*/,
   return Status::NotSupported("");
 }
 
-void CheckpointImpl::CleanStagingDirectory(const std::string& full_private_path,
-                                           Logger* info_log) {
+Status CheckpointImpl::CleanStagingDirectory(
+    const std::string& full_private_path, Logger* info_log) {
   std::vector<std::string> subchildren;
   Status s = db_->GetEnv()->FileExists(full_private_path);
   if (s.IsNotFound()) {
-    return;
+    // Nothing to clean
+    return Status::OK();
   }
   ROCKS_LOG_INFO(info_log, "File exists %s -- %s", full_private_path.c_str(),
                  s.ToString().c_str());
+
   s = db_->GetEnv()->GetChildren(full_private_path, &subchildren);
   if (s.ok()) {
     for (auto& subchild : subchildren) {
+      Status del_s;
       std::string subchild_path = full_private_path + "/" + subchild;
-      s = db_->GetEnv()->DeleteFile(subchild_path);
+      del_s = db_->GetEnv()->DeleteFile(subchild_path);
       ROCKS_LOG_INFO(info_log, "Delete file %s -- %s", subchild_path.c_str(),
                      s.ToString().c_str());
```

Comment:
We would want to log `del_s` (not `s`) here I think

Reply:
Good catch fixed